internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,335 @@
+	for {
+		attrs, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket.Objects: %v", err)
+		}
+		blobs = append(blobs, attrs.Name)
+	}
+	return blobs, nil
+}
+
+// getBlob gets the file reader of a blob in a Google Cloud Storage bucket.
+func getBlob(bucket *storage.BucketHandle, blobName string) (*storage.Reader, error) {
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", blobName, err)
+	}
+	defer reader.Close()
+	return reader, nil
+}
+
+// getRevisionFromFile gets the revision hash of the source code for which OSS-Fuzz coverage
+// reports were generated on a given day, given the file where the revision hash is saved.
+func getRevisionFromFile(content []byte, projectName string) (string, error) {
+	var payload map[string](map[string]string)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %v", err)
+	}
+	// Get the revisionHash using the file structure defined by OSS-Fuzz.
+	revisionHash := payload[fmt.Sprintf("/src/%s", projectName)]["rev"]
+	return revisionHash, nil
+}
+
+// parseCoverageSummary gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverageSummary(content []byte) (map[string]float64, map[string]float64, error) {
+	var summary CoverageSummary
+	err := json.Unmarshal(content, &summary)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
+	// Return branch coverage and line coverage using the coverage summary structure.
+	return summary.Data[0].Totals["branches"], summary.Data[0].Totals["lines"], nil
+}
+
+// getLogs gets the log-files list of a fuzz-target on a given day.
+func getLogs(date string, projectName string, fuzzTarget string, fuzzEngine string, sanitizer string) (*storage.BucketHandle, []string) {
+	// logsBucket is the ClusterFuzz Google Cloud Storage containing the fuzzers logs for a given project.
+	logsBucket := fmt.Sprintf("%s-logs.clusterfuzz-external.appspot.com", projectName)
+	bucket, err := getBucket(logsBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fuzzengine := strings.ToLower(fuzzEngine)
+	// relativePath is the relative path in the logsBucket where the logs of a given fuzz-target on a given day are saved.
+	relativePath := fmt.Sprintf("%s_%s_%s/%s_%s_%s/%s", fuzzEngine, projectName, fuzzTarget, fuzzengine, sanitizer, projectName, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return bucket, blobs
+}
+
+// getFuzzEffortFromFile gets the fuzzingEffort from a single log file.
+func getFuzzEffortFromFile(reader io.Reader, revisionHash string, projectName string) (int, float64) {
+	var revisionHashInLog string
+	var err error
+	var timeFuzz float64
+	var numTests int
+	caser := cases.Title(language.English)
+	scanner := bufio.NewScanner(reader)
```

Comment:
nit, feel free to ignore :) 

```suggestion
	lineScanner := bufio.NewScanner(reader)
```